src/pipeline/fit/FIT.groovy

```diff
@@ -16,26 +16,32 @@ def configControlInterfaceIp(String rackhd_dir){
     }
 }
 
-def run(String rackhd_dir, Object fit_configure){
-    withCredentials([
-        usernamePassword(credentialsId: 'ff7ab8d2-e678-41ef-a46b-dd0e780030e1',
-                         passwordVariable: 'SUDO_PASSWORD',
-                         usernameVariable: 'SUDO_USER')])
-    {
-        String group = fit_configure.getGroup()
-        String stack = fit_configure.getStack()
-        String log_level = fit_configure.getLogLevel()
-        String extra_options = fit_configure.getExtraOptions()
-        try{
-            sh """#!/bin/bash -ex
-            pushd $rackhd_dir/test
-            ./runFIT.sh -p $SUDO_PASSWORD -g "$group" -s "$stack" -v $log_level -e "$extra_options" -w $WORKSPACE
-            popd
-            """
-        } finally{
-            dir("$WORKSPACE"){
-                junit 'xunit-reports/*.xml'
-            }
+def run(String rackhd_dir, Object fit_configure, Map rackhd_host_cred=null){
+    String group = fit_configure.getGroup()
+    String stack = fit_configure.getStack()
+    String log_level = fit_configure.getLogLevel()
+    String extra_options = fit_configure.getExtraOptions()
+
+    if (rackhd_host_cred != null){
+        String rackhd_user = rackhd_host_cred["user"]
+        String rackhd_pass = rackhd_host_cred["password"]
+        sh """#!/bin/bash -ex
+        pushd $rackhd_dir/config
+        sed -i "s/\"username\": \"vagrant\"/\"username\": \"$rackhd_user\"/g" credentials_default.json
+        sed -i "s/\"password\": \"vagrant\"/\"password\": \"$rackhd_pass\"/g" credentials_default.json
+        popd
+        """
+    }
```

Comment:
Could you please add a method configFITCredential() and move line 25~34 to the method?

Reply:
good idea